cranelift/src/bugpoint.rs

```diff
@@ -732,7 +736,7 @@ impl Mutator for MergeBlocks {
             .as_slice(&func.dfg.value_lists)
             .iter()
             .cloned()
-            .zip(func.dfg.inst_variable_args(pred.inst).iter().cloned())
+            .zip(branch_args)
             .collect::<Vec<_>>()
```

Comment:
I think I'd have an easier time convincing myself that this is correct if it were structured a little differently. I believe it's correct as-is but I had to think hard to come to that conclusion.

```rust
// FIXME: maybe free the detached block param value list?
let block_params = func.dfg.detach_block_params(block).as_slice(&func.dfg.value_lists).to_vec();
assert_eq!(block_params.len(), branch_args.len());
for (block_param, arg) in block_params.into_iter().zip(branch_args) {
...
```

Reply:
I like this more -- I too had to stare a lot at the input to that `for` loop, and this refactoring cleans it up quite a bit!